File: CodeForWin/Patterns/Exercise 03/Solution.py
#!/usr/bin/env python3

def print_pattern(ROW, COLUMN):
    for i in range(ROW):
        for x in range(COLUMN):
            # Print all first row and last row
            if i == 1 or i == ROW:
                print("*", end = '')
            # Print first and last star in middle columns
            elif x == 1 or x == COLUMN:
                print("*", end = '')
            else:
                print(" ")

'''
            if i != 1 or i != ROW:
                for j in range(i, DIAGONALS):
                    print("*")
'''

def main():
    try:
        rows = input("Input row number: ")
        columns = input("Input column number: ")
        ROW = int(rows) + 1
        COLUMN = int(columns) + 1
        print_pattern(ROW, COLUMN)
    except ValueError:
        print()


if __name__ == '__main__':
    main()

File: CodeForWin/Patterns/Exercise 03/Solution.c
#include <stdio.h>

int main()
{
  /* User input row and column */
  int COLUMN, ROW;
  printf("Input row number: ");
  scanf("%d", &ROW);
  printf("Input column number: ");
  scanf("%d", &COLUMN);

  /* Print out pattern */
  printf("\n");
  for (int i = 1; i <= ROW; i++){
    for (int x = 1; x <= COLUMN; x++){
      /* Print first and last row */
      if (i == 1 || i == ROW){
        printf("*");
      } else if (x == 1 || x == COLUMN || i == x || x == (ROW - i + 1)){
        printf("*"); /* Print the first and last column */
      } else {
        printf(" ");
      }
    }
    printf("\n");
  }

  return 0;
}
